client/proposer/service_test.go

```diff
@@ -2,16 +2,79 @@ package proposer
 
 import (
 	"crypto/rand"
+	"fmt"
 	"math/big"
+	"os"
 	"testing"
 
+	"github.com/prysmaticlabs/prysm/client/syncer"
+
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/prysmaticlabs/prysm/client/database"
 	"github.com/prysmaticlabs/prysm/client/internal"
+	"github.com/prysmaticlabs/prysm/client/mainchain"
+	"github.com/prysmaticlabs/prysm/client/p2p"
 	"github.com/prysmaticlabs/prysm/client/params"
+	"github.com/prysmaticlabs/prysm/client/txpool"
+	"github.com/prysmaticlabs/prysm/client/types"
+	pb "github.com/prysmaticlabs/prysm/proto/sharding/v1"
+	logTest "github.com/sirupsen/logrus/hooks/test"
 )
 
+func TestProposeCollation(t *testing.T) {
```

Comment:
2 feedback here:
1.) we are not calling `proposeCollations `, can we make sure we call it and test that using channels? it's similar to what we did in simulator and sync tests
2.) we are not testing `(sizeOfCollation + int64(gethtx.Size())) > types.CollationSizelimit` Can we add a test scenario where the `collation` is right before full?

Reply:
1) hey so `proposer.Start()` actually spawns off a goroutine which calls `proposeCollations` , so the method is being tested. Maybe TestProposeCollation is not the best name for it since we are proposing, creating and saving collations to the shardDB. Its more like a roundtrip test. 
We can use channels to test the broadacast and subscription of transactions through the p2p feed. 
2) Alright I will add another test scenario for that